plugin.video.raitv/default.py

```diff
@@ -0,0 +1,525 @@
+        else:
+            icon =''
+        
+        title = re.findall("<p class=\"info\">([^<]+?)</p>", i)
+        if title:
+            title = title[0]
+        else:
+            title = ''
+
+        startTime = re.findall("<p class=\"time\">([^<]+?)</p>", i)
+        if startTime:
+            title = startTime[0] + " " + title
+
+        desc = re.findall("<p class=\"descProgram\">([^<]+?)</p>", i, re.S)
+        if desc:
+            desc= desc[0]
+        else:
+            desc=""
+        
+        videoUrl = re.findall('''data-href=['"]([^'^"]+?)['"]''', i)
+
+        if not videoUrl:
+            # programme is not available
+            liStyle = xbmcgui.ListItem(" [I]" + title + "[/I]", thumbnailImage = icon)
+            liStyle.setInfo("video", {})
+            addLinkItem({"mode": "nop"}, liStyle)
+        else:
+            videoUrl = videoUrl[0]
+            if not videoUrl.endswith('json'):
+                videoUrl = videoUrl + "?json"
+
+            liStyle = xbmcgui.ListItem(title, thumbnailImage = icon )
+            liStyle.setInfo("video", {})
+            addLinkItem({"mode": "play", "path_id": videoUrl}, liStyle)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+
+def show_replay_radio_epg(date, channelId):
+    xbmc.log("Showing EPG for " + channelId + " on " + date)
+    raiplayradio = RaiPlayRadio()
+    programmes = raiplayradio.getProgrammes(channelId.decode("utf-8"), date)
+    
+    for programme in programmes:
+        if not programme:
+            continue
+    
+        startTime = programme["timePublished"]
+        title = programme["name"]
+        
+        if programme["images"]["landscape"] != "":
+            thumb = raiplayradio.getThumbnailUrl(programme["images"]["square"])
+        elif programme["isPartOf"] and programme["isPartOf"]["images"]["square"] != "":
+            thumb = raiplayradio.getThumbnailUrl(programme["isPartOf"]["images"]["square"])
+        else:
+            thumb = raiplayradio.noThumbUrl
+        
+        if programme["hasAudio"]:
+            audioUrl = programme["pathID"]
+        else:
+            audioUrl = None
+        
+        if audioUrl is None:
+            # programme is not available
+            liStyle = xbmcgui.ListItem(startTime + " [I]" + title + "[/I]",
+                thumbnailImage=thumb)
+            liStyle.setInfo("audio", {})
+            addLinkItem({"mode": "nop"}, liStyle)
+        else:
+            liStyle = xbmcgui.ListItem(startTime + " " + title,
+                thumbnailImage=thumb)
+            liStyle.setInfo("audio", {})
+            addLinkItem({"mode": "play",
+                "path_id": audioUrl}, liStyle)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+    
+def show_ondemand_root():
+    raiplay = RaiPlay()
+    items = raiplay.getMainMenu()
+    for item in items:
+        if item["sub-type"] in ("RaiPlay Tipologia Page", "RaiPlay Genere Page", "RaiPlay Tipologia Editoriale Page" ):
```

Comment:
Localize these strings pls

Reply:
These strings are used to search in json, so they aren't string in GUI

Reply:
Disregard, marking as solved.